### PR TITLE
Add JAX-formatting documentation

### PR DESCRIPTION
**Pull request by alvarobartt**
## What's in this PR?

As a follow-up of #5522, I've created this entry in the documentation to explain how to use `.with_format("jax")` and why is it useful.

@lhoestq Feel free to drop any feedback and/or suggestion, as probably more useful features can be included there!

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Awesome thank you !

Could you also explain how to use certain types like ClassLabel, Image or Audio with jax ? You can get a lot of inspiration from the "Other feature types" section in the [PyTorch page](https://huggingface.co/docs/datasets/use_with_pytorch)

I also think it's be nice if this page had the same structure as the pytorch or tf ones, with sections named
- Dataset format
- N-dimensional arrays
- Other feature types
- Data loading

**Comment by alvarobartt** [comment]
> Awesome thank you !
> 
> Could you also explain how to use certain types like ClassLabel, Image or Audio with jax ? You can get a lot of inspiration from the "Other feature types" section in the [PyTorch page](https://huggingface.co/docs/datasets/use_with_pytorch)
> 
> I also think it's be nice if this page had the same structure as the pytorch or tf ones, with sections named
> 
>     * Dataset format
> 
>     * N-dimensional arrays
> 
>     * Other feature types
> 
>     * Data loading

Sure @lhoestq I'll do that later this afternoon whenever I'm done working! Thanks for the feedback as always 🤗

**Comment by alvarobartt** [comment]
Also, @lhoestq do you want me to elaborate more on the `## Data loading` section on how to use `datasets` to train a JAX model offering alternatives e.g. `Flax`, or do I keep it pure JAX? Thanks!

**Comment by lhoestq** [comment]
If you have a good example with `flax` it can also be helpful for users

**Comment by alvarobartt** [comment]
For now I think that probably it's not worth to add a `Flax` example, as train loops need to be done manually as in pure JAX, so probably the JAX example is enough. Anyway, let me know if you see something missing/incomplete/misleading/etc. and I'll update that ASAP 👍🏻

**Comment by alvarobartt** [comment]
P.S. I see that the `benchmark` action is being triggered on every PR, is it worth it? e.g. now I'm just editing the docs, so does it make any sense to trigger still the whole CI pipeline (including `benchmark`)? Just asking because in this PR for example it could be skipped.

**Comment by lhoestq** [comment]
> P.S. I see that the benchmark action is being triggered on every PR, is it worth it? e.g. now I'm just editing the docs, so does it make any sense to trigger still the whole CI pipeline (including benchmark)? Just asking because in this PR for example it could be skipped.

We could restrict it to PRs modifying files in src/ indeed ^^'

File: docs/source/use_with_jax.mdx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+>>> ds[0]["image"].shape
+torch.Size([512, 512, 3])
+>>> ds[0]
+{'image': DeviceArray([[[ 255, 255, 255],
+              [ 255, 255, 255],
+              ...,
+              [ 255, 255, 255],
+              [ 255, 255, 255]]], dtype=uint8)}
+>>> ds[:2]["image"].shape
+torch.Size([2, 512, 512, 3])
```
Comment (lhoestq) [inline docs/source/use_with_jax.mdx:113]:
```suggestion
(2, 512, 512, 3)
```

File: docs/source/use_with_jax.mdx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+</Tip>
+
+```py
+>>> from datasets import Dataset, Features, Image
+>>> images = ["path/to/image.png"] * 10
+>>> features = Features({"image": Image()})
+>>> ds = Dataset.from_dict({"image": images}, features=features)
+>>> ds = ds.with_format("jax")
+>>> ds[0]["image"].shape
+torch.Size([512, 512, 3])
```
Comment (lhoestq) [inline docs/source/use_with_jax.mdx:105]:
```suggestion
(512, 512, 3)
```

**Review by lhoestq — APPROVED** [review]
LGTM :)

**Comment by alvarobartt** [comment]
> LGTM :)

Cool thanks! My bad I didn't update those code blocks 🙃 Thanks for doing so before merge!

**Pull request merged**